### PR TITLE
no-cache flag, gopath tmp variable...

### PR DESCRIPTION
**Pull request by bbrodriges**
...minor format cleanup

I have replaced GOPATH rewinding by traversing to temp original GOPATH variable, added new flag which removes cache after vendoring (can be useful for containers) and made a few code formatting and cleanup (pointed by IDE).

### CODE REVIEW

**Comment by imikushin** [comment]
@bbrodriges What is your use case for trash that really needs the changes you're proposing? 

I needed a tool to be run at development time (not even at build time) with 2 cases in mind: to add libraries to ./vendor and to minimize unused vendored code. Auto-removing cache is not important for trash to have, and having stuff in the cache makes the vendoring phase much faster for later uses (because we often work with a bunch of the same libs from project to project), and you can always come up with your own cache dir (that will be created and populated by trash) and remove it afterwards.

File: trash.go [hunk 1]
```diff
@@ -487,10 +492,12 @@ func removeEmptyDirs(rootPackage string) error {
 	return nil
 }
 
-func cleanup(dir string) error {
-	gopath := path.Join(dir, "..", "..", "..", "..")
-	gopath = filepath.Clean(gopath)
-	os.Setenv("GOPATH", gopath)
+func cleanup(dir string, trashDir string, clearCache bool) error {
+    	cacheFolder := os.Getenv("GOPATH")
```
Comment (imikushin) [inline trash.go]: This really makes no sense to me: cacheFolder is **not** the GOPATH: not in the cleanup stage. Even if it was, you don't need to pass it through environment. BTW, you've just added `trashDir` argument which is not used: why?

At this point we're assuming the project's GOPATH, hence the `../../../..`

**Comment by imikushin** [comment]
@bbrodriges Looks like I'm going to close this PR, but I really want to know how you're using trash and your needs as a user, so welcome to comment.

**Comment by bbrodriges** [comment]
Hi! I was thinking to use trash in Docker image building process. I don't really want to push vendor folder to my git repo. So "no-cache" flag is just a handy shortcut for rming trash folder later, since I don't need it after my app build anymore.

As of GOPATH^  I can clearly see that trash changes it on line 120 to trashDir (which equals ```c.String("cache")```) and not changing it afterwards. Because of it I had error ```Error removing unused dirs``` as GOPATH was wrong after traversing.

I now can clearly see that it was bad idea to store original GOPATH under temporary env var, it is more convenient to store it under global package variable. I must have been very tired and forget to remove unused trashDir param.

**Pull request closed without merging**